### PR TITLE
Update custom policies for Symfony 5 compliance.

### PR DESCRIPTION
**Pull request by rhidja**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | NA
| Versions      | 3.1

### CODE REVIEW

File: docs/guide/custom_policies.md [hunk 1]
```diff
@@ -118,6 +118,32 @@ class AcmeExampleBundle extends Bundle
 }
 ```
 
+If you are using Symfony 5 you have to use the `App\Kernel` to register a new policy provider :
```
Comment (andrerom) [inline docs/guide/custom_policies.md:121]: as this is targeting 3.1 and higher, do we need to mention things like `if you are suing sf 5` ?

Comment (rhidja) [reply docs/guide/custom_policies.md]: Can replace Symfony 5, EzPlatform 3.1

**Comment by andrerom** [comment]
@dabrt Change should rather replace the way it was done before which is no longer valid _(**if** that is the case)_, rather then a If / else depending on version _(already covered by branching)_.

File: docs/guide/custom_policies.md [hunk 2]
```diff
@@ -118,6 +118,32 @@ class AcmeExampleBundle extends Bundle
 }
 ```
 
+If you are using EzPlatform 3.0+ you have to use the `App\Kernel` to register a new policy provider :
```
Comment (dabrt) [inline docs/guide/custom_policies.md:121]:
```suggestion
If you are using EzPlatform 3.0+ you must use the `App\Kernel` to register a new policy provider:
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/guide/custom_policies.md (hunk 1 above)
Comment (andrerom) [reply docs/guide/custom_policies.md]: That is not my point, your suggestion towards 3.1 doc will result in 3.1 doc containing two instructions for how to do this. One which you seems to indicate does not work on 3.1 and then a second way for 3.1

Summary: 3.1 branch is 3.1 doc => 3.1 doc only need instructions for how you do things on 3.1, we don't need both here.

**Review by dabrt — APPROVED** [review]

**Comment by DominikaK** [comment]
@rhidja , I've had to modify branch history, can you please rebase this PR against the current 3.1 branch?

**Comment by rhidja** [comment]
@DominikaK rebase done!

**Comment by andrerom** [comment]
@dabrt thanks for review ping, but I can't review as long as I don't quite get why we don't just adjust the code example above.

As in the change here seems to indicate the way to do it changed with 3.x, so why keep doc on how it's done in earlier releases? Or is there some reason to show both approaches?

**Comment by rhidja** [comment]
@andrerom @dabrt  il removed the policy definion for old ezpaltform versions.

File: docs/guide/custom_policies.md [hunk 3]
```diff
@@ -125,8 +126,7 @@ To provide support for editing custom policies in the Back Office, you need to i
 Next, register the service in DIC (Dependency Injection Container) with the `ez.limitation.formMapper` tag and set the `limitationType` attribute to the Limitation type's identifier:
```
Comment (andrerom) [inline docs/guide/custom_policies.md:125]: Not part of the changes here but, @DominikaK we kind of stopped shortening Dependency Injection Container didn't we?

**Review by andrerom — APPROVED** [review]

File: docs/guide/custom_policies.md (hunk 3 above)
Comment (DominikaK) [reply docs/guide/custom_policies.md:125]: If you're referring to using the DIC acronym, then yes, we need to unify the way we refer to registering services (in most places we just say "register the service..." without spelling out DIC). To be done in some other PR throughout the whole doc.

Comment (rhidja) [reply docs/guide/custom_policies.md:125]: @andrerom, @DominikaK done.

**Review by mikadamczyk — APPROVED** [review]

File: docs/guide/custom_policies.md [hunk 4]
```diff
@@ -94,26 +94,27 @@ It is not possible to remove an existing module, function or limitation from a P
...
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    {
         // ...
- 
+        
         // Retrieve "ezpublish" container extension
         $eZExtension = $container->getExtension('ezpublish');
         // Add the policy provider
-        $eZExtension->addPolicyProvider(new MyPolicyProvider());
+        $eZExtension->addPolicyProvider(new AppPolicyProvider());
```
Comment (micszo) [inline docs/guide/custom_policies.md:117]: It looks like now it's inconsistent, L43 & L69 still use "MyPolicyProvider".

File: docs/guide/custom_policies.md [hunk 5]
```diff
@@ -122,11 +123,10 @@ class AcmeExampleBundle extends Bundle
 
 To provide support for editing custom policies in the Back Office, you need to implement [`EzSystems\EzPlatformAdminUi\Limitation\LimitationFormMapperInterface`](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/lib/Limitation/LimitationFormMapperInterface.php).
```
Comment (micszo) [inline docs/guide/custom_policies.md:124]: Some example code here would be really useful. 🙂 It's rather a general statement.

File: docs/guide/custom_policies.md [hunk 6]
```diff
@@ -135,11 +135,10 @@ app.security.limitation.custom_limitation.mapper:
 
 If you want to provide human-readable names of the custom Limitation values, you need to implement [`EzSystems\EzPlatformAdminUi\Limitation\LimitationValueMapperInterface`](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/lib/Limitation/LimitationValueMapperInterface.php).
```
Comment (micszo) [inline docs/guide/custom_policies.md:136]: Here also some example would be useful.

**Review by mikadamczyk — COMMENTED** [review]
It would be great to add information to the doc about how to create custom limitation e.g. CustomLimitationType.php and CustomLimitation.php classes with sample configuration 
```
    Security\Limitation\CustomLimitationType:
        tags:
            - {name: ezpublish.limitationType, alias: CustomLimitation}
``` 
to make it more clear

File: docs/guide/custom_policies.md (hunk 4 above)
Comment (rhidja) [reply docs/guide/custom_policies.md]: @micszo done!

File: docs/guide/custom_policies.md (hunk 5 above)
Comment (rhidja) [reply docs/guide/custom_policies.md:124]: @micszo it is too long to do it, i sugest to merge first this PR, to correct the current documentation. After i think it is better to split this chapter, and create an other one for custom limitations, to explain how to create CustomLimitation, CustomLimitationType,  LimitationFormMapperInterface, imitationValueMapperInterface.

File: docs/guide/custom_policies.md (hunk 6 above)
Comment (rhidja) [reply docs/guide/custom_policies.md:136]: @micszo same thing too long to do it.

File: docs/guide/custom_policies.md (hunk 5 above)
Comment (micszo) [reply docs/guide/custom_policies.md:124]: Yes, internally we also came to the conclusion that this will have to be tackled separately. \cc @dabrt

**Review by micszo — APPROVED** [review]
LGTM. 👍 

Let's handle the big issues separately.

File: docs/guide/custom_policies.md (hunk 5 above)
Comment (dabrt) [reply docs/guide/custom_policies.md:124]: A ticket was created to handle your suggestions to this article:
https://issues.ibexa.co/browse/EZP-32155
Should you have any more comments, you can add them directly there.

Comment (micszo) [reply docs/guide/custom_policies.md:124]: Thanks!

**Pull request merged**